extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/DependenciesProcessor.java

```diff
@@ -0,0 +1,430 @@
+            }
+            if (getClass() != obj.getClass()) {
+                return false;
+            }
+            final Link other = (Link) obj;
+            if (!Objects.equals(this.source, other.source)) {
+                return false;
+            }
+            if (!Objects.equals(this.target, other.target)) {
+                return false;
+            }
+            return Objects.equals(this.type, other.type);
+        }
+    }
+
+    private static DepNode getTargetDepNode(ApplicationModel model, ArtifactCoords targetCoords) {
+        DepNode targetDep = getTargetWithAllDependents(targetCoords, getCompleteMap(model));
+        // It seems like all the possible paths look like a lot in many cases.
+        // Common shortest paths is an attempt to reduce the number of paths displayed
+        // by selecting all the unique shortest paths
+        final boolean showShortestCommonPaths = false;
+        if (showShortestCommonPaths) {
+            final List<DepPath> paths = collectAllPaths(targetDep);
+            filterShortestCommonPaths(paths);
+            return getGraph(paths);
+        }
+        return targetDep;
+    }
+
+    private static DepNode getGraph(List<DepPath> paths) {
+        final DepNode newTarget = new DepNode(paths.get(0).path.get(0).id, paths.get(0).path.get(0).resolvedDep);
+        final Map<ArtifactKey, DepNode> nodes = new HashMap<>();
+        for (var path : paths) {
+            var currentNode = newTarget;
+            for (int i = 1; i < path.path.size(); ++i) {
+                var originalNode = path.path.get(i);
+                var newNode = nodes.computeIfAbsent(originalNode.resolvedDep.getKey(),
+                        k -> new DepNode(originalNode.id, originalNode.resolvedDep));
+                currentNode.addDependent(newNode);
+                currentNode = newNode;
+            }
+        }
+        return newTarget;
+    }
+
+    private static void filterShortestCommonPaths(List<DepPath> paths) {
+        int i = 0;
+        while (i < paths.size()) {
+            var currentPath = paths.get(i++);
+            int j = i;
+            while (j < paths.size()) {
+                if (paths.get(j).ids.containsAll(currentPath.ids)) {
+                    paths.remove(j);
+                } else {
+                    ++j;
+                }
+            }
+        }
+        logPaths(paths, "Shortest common paths");
+    }
+
+    private static List<DepPath> collectAllPaths(DepNode targetDep) {
+        final List<DepPath> allPaths = new ArrayList<>();
+        collectAllPaths(targetDep, new DepPath(), allPaths);
+        Collections.sort(allPaths);
+        logPaths(allPaths, "Paths");
+        return allPaths;
+    }
+
+    private static void logPaths(List<DepPath> paths, String title) {
+        System.out.println(title + " total: " + paths.size());
+        int j = 1;
+        for (var dp : paths) {
+            System.out.println(j++ + ") " + dp);
+        }
+    }
```

Comment:
Maybe this should be removed ?

Reply:
This will be logged only if the shortest common paths are enabled, which is for now hardcoded to false.